extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs

```diff
@@ -0,0 +1,272 @@
+            Type _ when targetType == typeof(IEnumerable<AppendBlobClient>) => blobCollection.Select(b => (AppendBlobClient)b),
+            Type _ when targetType == typeof(IEnumerable<BlobContainerClient>) => blobCollection.Select(b => (BlobClient)b),
+            _ => null
+        };
+
+        private object[]? ToTargetTypeArray(Type targetType, IEnumerable<object> blobCollection) => targetType switch
+        {
+            Type _ when targetType == typeof(BlobBaseClient[]) => blobCollection.Select((b) => ((BlobBaseClient)b)).ToArray(),
+            Type _ when targetType == typeof(BlobClient[]) => blobCollection.Select((b) => ((BlobClient)b)).ToArray(),
+            Type _ when targetType == typeof(String[]) => blobCollection.Select((b) => ((String)b)).ToArray(),
+            Type _ when targetType == typeof(Stream[]) => blobCollection.Select((b) => ((Stream)b)).ToArray(),
+            Type _ when targetType == typeof(Byte[][]) => blobCollection.Select((b) => ((Byte[])b)).ToArray(),
+            Type _ when targetType == typeof(BlockBlobClient[]) => blobCollection.Select((b) => ((BlockBlobClient)b)).ToArray(),
+            Type _ when targetType == typeof(PageBlobClient[]) => blobCollection.Select((b) => ((PageBlobClient)b)).ToArray(),
+            Type _ when targetType == typeof(AppendBlobClient[]) => blobCollection.Select((b) => ((AppendBlobClient)b)).ToArray(),
+            Type _ when targetType == typeof(BlobContainerClient) => blobCollection.Select((b) => ((BlobClient)b)).ToArray(),
+            _ => null
+        };
+
+        private async Task<object?> DeserializeToTargetObject(Type targetType, string connectionName, string containerName, string blobName)
+        {
+            var content = await GetBlobStream(connectionName, containerName, blobName);
+            return _workerOptions.Value.Serializer.Deserialize(content, targetType, CancellationToken.None);
+        }
+
+        private object? DeserializeToTargetObjectCollection(IEnumerable<object> blobCollection, Type targetType)
+        {
+            (string methodName, Type type) = targetType.IsArray ? (nameof(CloneToArray), targetType.GetElementType())
+                                                            : (nameof(CloneToList), targetType.GenericTypeArguments[0]);
+
+            blobCollection = blobCollection.Select(b => Convert.ChangeType(b, type));
+            MethodInfo method = typeof(BlobStorageConverter).GetMethod(methodName);
+            MethodInfo genericMethod = method.MakeGenericMethod(type);
+
+            return genericMethod.Invoke(null, new[] {blobCollection.ToList()});
+        }
+        
+        public static T[] CloneToArray<T>(IList<object> source)
+        {
+            var result = source.Cast<T>().ToList();
+            return result.ToArray();
+        }
+
+        public static IEnumerable<T> CloneToList<T>(IList<object> source)
+        {
+            return source.Cast<T>().ToList();
+        }
```

Comment:
Wondering if we should create a utilities sort of file for things like this, I think this class is getting rather large and breaking things down might help us better maintain this. Esp if we might be adding output binding support

Reply:
You could if you want, but the way our extension files are structured would require a shared internals project. Might be more work than it is worth. Also, just inlining `*.Cast<T>().ToList()` instead of an extension method isn't the worst either.

Reply:
I think the extension method currently exists so it can be called by reflection, any ideas on how we can improve that?

```csharp
        private object? DeserializeToTargetObjectCollection(IEnumerable<object> blobCollection, Type targetType)
        {
            (string methodName, Type type) = targetType.IsArray
                                            ? (nameof(CloneToArray), targetType.GetElementType())
                                            : (nameof(CloneToList), targetType.GenericTypeArguments[0]);

            blobCollection = blobCollection.Select(b => Convert.ChangeType(b, type));
            MethodInfo method = typeof(BlobStorageConverter).GetMethod(methodName);
            MethodInfo genericMethod = method.MakeGenericMethod(type);

            return genericMethod.Invoke(null, new[] { blobCollection.ToList() });
        }
```

Reply:
@jviau - just in case you missed this!

Reply:
Ah I see how these are used. I can't really think of anything better off the top of my head. It would be nice to move these to a shared extension. I'll start an offline chat with the team to see how we want to do that.